WatchDogEclipsePlugin/WatchDog/src/nl/tudelft/watchdog/eclipse/ui/handlers/StartupUIThread.java

```diff
@@ -56,19 +56,26 @@ public void run() {
 
 	/** Checks whether there is a registered WatchDog user */
 	private void checkWhetherToDisplayUserProjectRegistrationWizard() {
-		if (!WatchDogUtils.isEmpty(preferences.getUserid()))
+		ProjectPreferenceSetting projectSetting = preferences
+				.getOrCreateProjectSetting(workspaceName);
+		if (!WatchDogUtils.isEmpty(preferences.getUserid())
+				|| (projectSetting.startupQuestionAsked && !projectSetting.enableWatchdog))
 			return;
 
 		UserRegistrationWizardDialogHandler newUserWizardHandler = new UserRegistrationWizardDialogHandler();
 		try {
 			int statusCode = (int) newUserWizardHandler
 					.execute(new ExecutionEvent());
+			savePreferenceStoreIfNeeded();
 			if (statusCode == Window.CANCEL) {
 				if (MessageDialog.openQuestion(null, "WatchDog not active!",
```

Comment:
Can you factor out the call to the MessageDialog to make it more apparent what is going on here?